glide-core/src/client/reconnecting_connection.rs

```diff
@@ -268,4 +307,14 @@ impl ReconnectingConnection {
             ConnectionState::Reconnecting
         )
     }
+
+    pub async fn wait_for_disconnect_with_timeout(&self, max_wait: &Duration) {
+        // disconnect_notifier always exists
+        self.connection_options
+            .disconnect_notifier
+            .as_ref()
+            .unwrap()
```

Comment:
If you must use it, then use it like this:

```rust
let Some(notifier) = &self.connection_options.disconnect_notifier {
   tracing::error!("No notifier!");
   unreachable!();
};

notider. ....
```

Reply:
Fixed